src/System.Numerics.Tensors/System/Numerics/Tensor.cs

```diff
@@ -35,7 +35,8 @@ public static Tensor<T> CreateIdentity<T>(int size, bool columMajor)
         /// <returns></returns>
         public static Tensor<T> CreateIdentity<T>(int size, bool columMajor, T oneValue)
         {
-            var result = new DenseTensor<T>(new[] { size, size }, columMajor);
+            var dimensions = new[] { size, size }.AsReadOnlySpan();
```

Comment:
Consider re-writing as follows since it is clearer that dimensions is a Span of int.
```C#
ReadOnlySpan<int> dimensions = new[] {size, size};
```

Reply:
I'll use @KrzysztofCwalina's stackalloc suggestion.

Reply:
You can stackalloc, but you cannot currently initialize the data inline.